bzt/modules/python.py

```diff
@@ -307,14 +318,45 @@ def gen_setup(self, test_method):
         test_method.append(self.gen_new_line(indent=0))
 
     def gen_setup_method(self):
+        desire_capabilities = {}
+        inherited_capabilities = []
+
         self.log.debug("Generating setUp test method")
-        browsers = ["Firefox", "Chrome", "Ie", "Opera"]
-        browser = self.scenario.get("browser", "Firefox")
-        if browser not in browsers:
+        browsers = ["Firefox", "Chrome", "Ie", "Opera", "Remote", "Android", "iOS"]
+
+        browser = dict(self.scenario).get("browser", None)
+        # Split platform: Browser
+        browser_split = []
+        if browser:
+            browser_split = browser.split("-")
+            browser = browser_split[0]
+        if browser and (browser not in browsers):
             raise TaurusConfigError("Unsupported browser name: %s" % browser)
 
         setup_method_def = self.gen_method_definition("setUp", ["self"])
 
+        remote_executor = dict(self.scenario).get("remote", None)
+        self.log.info("Exec Remote:" + str(remote_executor))
+
+        if not browser and remote_executor:
+            browser = "Remote"
+        elif browser and browser in ["Android", "iOS"]:
```

Comment:
this is very confusing, you are using browser as platform. Instead, the naming convention could be "Chrome-Android", so browser is always first word and platform is optional second part. Also, parse it in the beginning into two clear variables, platform=None if it's not defined. So there won't be need to drag unobvious array and check for its lenght later.

Reply:
I just understood what you meant with platform.
I am releasing some changes to have a cleaner code, it will support the following models
- Chrome-Android
- Safari-iOS
- Native-Android
- Native-iOS
Native, following the convention of the client first, then the platform.